docs/contributors/code/testing-overview.md

```diff
@@ -334,21 +335,27 @@ Snapshots themselves don't express anything about what we expect. Snapshots are
 
 ```js
 test( 'should contain mars if planets is true', () => {
-	const wrapper = shallow( <SolarSystem planets /> );
+	const { container, getByText } = render( <SolarSystem planets /> );
 
 	// Snapshot will catch unintended changes
-	expect( wrapper ).toMatchSnapshot();
+	expect( container.firstChild ).toMatchSnapshot();
 
 	// This is what we actually expect to find in our test
-	expect( wrapper.find( Mars ) ).toHaveLength( 1 );
+	expect( getByText( /mars/i ) ).toBeInTheDocument();
 } );
 ```
 
-[`shallow`](http://airbnb.io/enzyme/docs/api/shallow.html) rendering is your friend:
-
-> Shallow rendering is useful to constrain yourself to testing a component as a unit, and to ensure that your tests aren't indirectly asserting on behavior of child components.
+Another good technique is to use the `toMatchDiffSnapshot` function (provided by the [`snapshot-diff` package](https://github.com/jest-community/snapshot-diff)), which allows to snapshot only the difference between two different states of the DOM. This approach is useful to test the effects of a prop change on the resulting DOM while generating a much smaller snapshot, live in this example:
 
-It's tempting to snapshot deep renders, but that makes for huge snapshots. Additionally, deep renders no longer test a single component, but an entire tree. With `shallow`, we snapshot just the components that are directly rendered by the component we want to test.
+```js
+test( 'should render a darker background when isShady is true', () => {
```

Comment:
```suggestion
```jsx
test( 'should render a darker background when isShady is true', () => {
```

Reply:
I don't know if the handbook supports ```jsx. In case it doesn't, just ignore these suggestions.

Reply:
Agreed. Done in [`e5de664` (#34423)](https://github.com/WordPress/gutenberg/pull/34423/commits/e5de6647a9dfc4912aac29edbb2f0ca3ae6b6524)